woocommerce/woocommerce
# dev: added xstate v5 inspector

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This PR adds the Stately AI inspector for XState V5 machines since the old @xstate/inspect package does not work for V5.

Additionally, since the V4 and V5 inspectors conflict, there's been some addition to determine which one should be enabled.

Closes #https://github.com/woocommerce/team-ghidorah/issues/296



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Run woocommerce application
2. For V4 machines, you need `localStorage.xstate_inspect = true`
3. For V5 machines you need both `localStorage.xstate_inspect = true` and `localStorage.xstateV5_inspect = true`
4. To disable the inspector you need `localStorage.xstate_inspect = false`
5. Test around the XState machines and see that they work as expected. Here are some URLs:
    - (V5) http://localhost:3000/wp-admin/admin.php?page=wc-admin&path=%2Flaunch-your-store
    - (V4) http://localhost:3000/wp-admin/admin.php?page=wc-admin&path=%2Fsetup-wizard
    - (V4) http://localhost:3000/wp-admin/admin.php?page=wc-admin&path=%252Fcustomize-store



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
